### ISSUE TITLE
Add custom RAW format conversion tool support

### ISSUE THREAD

**Issue by fierceX** [photoview/photoview#928]
Previously, I tried to use `Darktable` for `RAW` format conversion, but it was a pity that I could not install `Darktable` on my machine. Later, I found another alternative tool, `ImageMagick`, but I found that it did not solve my problem. Since most of the RAW formats I shot were dark (I only have one camera, I am not sure if the other `RAW` formats are the same), I need to increase the exposure during the conversion so that it looks normal. But when I use `ImageMagick` to adjust the exposure (actually brightness), some of the highlights will overflow, making it difficult for me to adjust automatically. Then I found a library for decoding RAW format: [libraw](https://www.libraw.org/), which has the option of analog exposure adjustment, and can retain highlights to avoid highlight overflow when adjusting exposure. Unfortunately, the library does not have an executable tool, so I packaged a raw2jgp tool myself, which now solves my problem and enables my `RAW` photos to adjust their exposure during conversion. And there will be no highlight overflow. From this, I wonder if it is possible to add a custom environment variable of the `RAW` conversion tool, and the conversion script can be specified by the user, so that various conversion needs can be met.

Here is my PR:
#929

**Comment by googollee** [comment]
Hi @fierceX. I have some questions about your workflow. As you said you are using `raw2jpg` and adjusting exposure during the conversion. It sounds like you are using `photoview` to edit images in batch. Are you using the same parameters to convert all images across the time, or do you change some parameters each time before conversion? Thank you.

**Comment by fierceX** [comment]
@googollee 
I'm converting images in batch, and currently I can only convert them with the same parameters, I'd prefer to have a button in the interface to re-convert the images, as well as customize the incoming parameters。
Thinking about that direction now?

**Comment by googollee** [comment]
@fierceX Actually, not that direction. `photoview` uses `darktable` to generate thumbnails and web images but not for conversion, as it's a gallery but not a editor.

For your use case, I suggest to convert your images with scripts, store jpg files with the same name together with original files. `photoview` will scan those files and stack images with the same name, and show jpg files online.

**Comment by kkovaletp** [comment]
@fierceX, as part of the #1011 discussion we discovered a big security issue in providing users a way to configure custom executables for workers: they are going to be launched on the server by the scanner job automatically. So, we have to make sure that all of them are safe to run, which cannot be guaranteed if a user can redefine them.
There is another issue from the quality of the results PoV: users might provide incorrect/incompatible/not optimal parameters to those tools, and get unexpected or poor-quality results, which impacts the overall impression from Photoview and will trigger a lot of support cases.

So, we decided for now to keep the list of executables, integrated with Photoview, under the dev team's control, at least for now. And we're in the process of migrating to ImageMagick and a better build of FFmpeg. We expect that it will give us many benefits.

With this, I'm closing this PR as not planned. Anyway, you did a great job! Thanks for bringing this topic to us and participating in the discussion!
